src/main/java/com/thealgorithms/sorts/DarkSort.java

```diff
@@ -0,0 +1,69 @@
+package com.thealgorithms.sorts;
+
+/**
+ * Dark Sort algorithm implementation.
+ *
+ * Dark Sort uses a temporary array to count occurrences of elements and
+ * reconstructs the sorted array based on the counts.
+ *
+ * @see SortAlgorithm
+ */
+class DarkSort implements SortAlgorithm {
+
+    /**
+     * Sorts the array using the Dark Sort algorithm.
+     *
+     * @param unsorted the array to be sorted
+     * @param <T> Comparable class
+     * @return sorted array
+     */
+    @Override
+    public <T extends Comparable<T>> T[] sort(T[] unsorted) {
```

Comment:
Could you remove default SortAlgorithm interface and just accept Integer[]? We don't have an appropriate interface for integer sorts. The current implementation violates Liskov Substitution Principle